### PR TITLE
Fix inaccurate MySQL/MariaDB, Redis metrics

### PR DESCRIPTION
**Pull request by yswdqz**
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).
snapshots: 
c30cc23a22a62208ef61e472b0f9bb53
5d8920f04897a80db288239eca2e86bc

### CODE REVIEW

File: test/e2e-v2/cases/mariadb/expected/metrics-has-value-metric-null.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+type: TIME_SERIES_VALUES
+results:
+  {{- contains .results }}
+  - metric: null
```
Comment (wankai123) [inline test/e2e-v2/cases/mariadb/expected/metrics-has-value-metric-null.yml:34]: why `null` here and what's this expected file for?

**Comment by wu-sheng** [comment]
I think those service dashboards should be updated to use MQE to aggregate instance metrics.

**Comment by wu-sheng** [comment]
Redis e2e fails, please fix it.

File: test/e2e-v2/cases/mariadb/expected/metrics-has-value-metric-null.yml (hunk 1 above)
Comment (wankai123) [reply test/e2e-v2/cases/mariadb/expected/metrics-has-value-metric-null.yml:34]: OK, I think I know why `null` of the meta here, I'll fix it later.

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/mysql/mysql-service.json [hunk 2]
```diff
@@ -292,32 +292,8 @@
       "layer": "MYSQL",
       "entity": "Service",
       "name": "MySQL-Service",
-      "isRoot": false,
-      "isDefault": true,
-      "expressions": [
-        "avg(meter_mysql_commands_select_rate)",
-        "avg(meter_mysql_commands_insert_rate)",
-        "avg(meter_mysql_commands_update_rate)",
-        "avg(meter_mysql_commands_delete_rate)"
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/mysql/mysql-service.json:301]: Why are these removed?

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/postgresql/postgresql-service.json [hunk 3]
```diff
@@ -448,31 +448,7 @@
       "layer": "POSTGRESQL",
       "entity": "Service",
       "name": "PostgreSQL-Service",
-      "isDefault": true,
-      "expressions": [
-        "avg(meter_pg_fetched_rows_rate)",
-        "avg(meter_pg_inserted_rows_rate)",
-        "avg(meter_pg_updated_rows_rate)",
-        "avg(meter_pg_deleted_rows_rate)"
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/postgresql/postgresql-service.json:456]: Same here

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/mysql/mysql-service.json (hunk 2 above)
Comment (wankai123) [reply oap-server/server-starter/src/main/resources/ui-initialized-templates/mysql/mysql-service.json:301]: @yswdqz Are you using the latest UI version?  It shouldn't delete the hierarchy part when export.

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/postgresql/postgresql-service.json [hunk 4]
```diff
@@ -31,7 +31,7 @@
                     "title": "Fetched Rows Trend (rows per second)"
                   },
                   "expressions": [
-                    "meter_pg_fetched_rows_rate"
+                    "aggregate_labels(meter_pg_fetched_rows_rate,sum(service_instance_id))"
```
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/postgresql/postgresql-service.json:34]: Do you want to show instance metrics separately?

Comment (wu-sheng) [reply oap-server/server-starter/src/main/resources/ui-initialized-templates/postgresql/postgresql-service.json:34]: This is service dashboard, you should merge instance tag to aggregate.

Comment (yswdqz) [reply oap-server/server-starter/src/main/resources/ui-initialized-templates/postgresql/postgresql-service.json:34]: Yes, I put details in instance level, and service only display different instances metrics.

Comment (wu-sheng) [reply oap-server/server-starter/src/main/resources/ui-initialized-templates/postgresql/postgresql-service.json:34]: > Yes, I put details in instance level, and service only display different instances metrics.

This isn't the point of this fix. The key to do this is making service metric back. If only showing per instance, the old kernel works already.

Please fix accordingly.

**Comment by wankai123** [comment]
please check and help fix the hierarchy part: https://github.com/apache/skywalking/blob/7e978261fcb2bf2f619b8f2a9586e1703107f0a3/oap-server/server-starter/src/main/resources/ui-initialized-templates/mysql/mysql-service.json#L301

**Comment by wankai123** [comment]
you didn't add `instance_id` in these expressions: 
https://github.com/apache/skywalking/blob/7e978261fcb2bf2f619b8f2a9586e1703107f0a3/oap-server/server-starter/src/main/resources/otel-rules/mysql/mysql-service.yaml#L37

File: oap-server/server-starter/src/main/resources/otel-rules/mysql/mysql-service.yaml [hunk 5]
```diff
@@ -59,8 +59,10 @@ metricsRules:
...
   - name: connects_aborted
     exp: mysql_global_status_aborted_connects
-  - name: connects_available
-    exp: mysql_global_variables_max_connections.sum(['host_name']) - mysql_global_status_threads_connected.sum(['host_name'])
+  - name: max_connections
+    exp: mysql_global_variables_max_connections.sum(['host_name','service_instance_id'])
+  - name: status_thread_connected
+    exp: mysql_global_status_threads_connected.sum(['host_name','service_instance_id'])
   - name: connection_errors_max_connections
     exp: mysql_global_status_connection_errors_total.tagEqual('error','max_connection')
```
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/otel-rules/mysql/mysql-service.yaml:67]: Why do some of the metrics do not require `service_instance_id` ? They are not instance level metrics?

**Comment by yswdqz** [comment]
Got it, sorry I don't have time tonight, I'll fix it tomorrow.

File: oap-server/server-starter/src/main/resources/otel-rules/mysql/mysql-service.yaml [hunk 6]
```diff
@@ -44,7 +44,7 @@ metricsRules:
   - name: qps
     exp: mysql_global_status_queries.rate('PT1M')
   - name: tps
-    exp: mysql_global_status_commands_total.tagMatch('command','rollback|commit').sum(['host_name']).rate('PT1M')
+    exp: mysql_global_status_commands_total.tagMatch('command','rollback|commit').sum(['host_name', 'service_instance_id']).rate('PT1M')
```
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/otel-rules/mysql/mysql-service.yaml:47]: don't forget to add `service_instance_id` label in the e2e's OTEL config.

**Comment by yswdqz** [comment]
This PR is not yet complete (the UI part has not been modified). I'm very sorry as I had many things at school this weekend and didn't have time to work on it. I will definitely finish this PR tomorrow.

**Comment by wu-sheng** [comment]
Please fix the relative e2e.

File: test/e2e-v2/cases/mariadb/expected/metrics-has-value-metric-null.yml (hunk 1 above)
Comment (wankai123) [reply test/e2e-v2/cases/mariadb/expected/metrics-has-value-metric-null.yml:34]: @yswdqz https://github.com/apache/skywalking/pull/12142 make metadata not return null, you should fix this e2e

**Comment by yswdqz** [comment]
PR is ready if e2e pass.

**Review by wankai123 — APPROVED** [review]
